handbook/network-security/network-security.md

```diff
@@ -0,0 +1,225 @@
+
+### HTTPS in NI Web Server
+
+NI Web Server supports TLS 1.2.
+
+**NI Web Server Configuration** supports creating self-signed certificates, certificate signing requests (CSR), and installing certificates generated by a certificate authority. The DNS settings for the server can affect the operability of these certificates. Refer to [**DNS Configuration**](#dns-configuration) for details.
+
+!!! warn
+    Self signed certificates should be used for testing purposes only.
+
+!!! note "Limited capabilities when using self-signed certificates"
+
+    When using a self-signed certificates, clients will not trust the  by default. This can prevent some operations from occurring.
+
+    - Web browsers will not trust the certificate and the user must grant an exception to load the SystemLink web application.
+
+    - Packages built in LabVIEW cannot be [automatically published](https://www.ni.com/documentation/en/systemlink/latest/deployment/creating-packages-labview-package-builder/) to SystemLink server.
+
+    - The LabVIEW Client API will have to explicitly disable verify server checks to communicate with the server.
+        - Refer to [Open Configuration ](https://www.ni.com/documentation/en/systemlink/latest/systemlink-labview-node-ref/open-configuration-http-auto/) to details on this setting.
+
+When you have received a certificate from an certificate signing authority, you can use **NI Web Server Configuration** to install the certificate.
+
+1. Open **NI Web Sever Configuration** and navigate to the **HTTPS** tab.
+
+2. Click the **Use a certificate from a signing authority** radio button.
+
+3. Expand the **Install an already signed certificate** section.
+
+4. Click the folder icon next to **certificate file** and browse to your certificate file.
+
+5. Click the folder icon next to **Key file** and select your key file.
+
+6. If needed change the HTTPS port from the default, 443.
+
+7. Click **Apply and restart**.
+
+!!! note "TLS certificates with application load balancers"
+    If you have a load balancer in front of your SystemLink app server you must ensure the same certificate is installed on both the load balancer and NI Web Server for targets to successfully connect and publish data to SystemLink.
+
+### DNS Configuration
+
+SystemLink and NI Web Server do not ship with a DNS server. NI presumes you will provide a DNS server to be used in your environment. NI Web Server will attempt to provide a default for the DNS of your SystemLink server based on Windows OS settings. Other valid DNS names are listed in the **Preferred host name for generated URLs and certificates** combo box. You may also manually additional hostnames as needed. As the name of the combo box implies, this setting must match the host name in your TLS certificates. If an invalid hostname is provided data from managed targets will not be published.
```

Comment:
OpenID Connect provider redirect uri configuration must match DNS name as well.

Caveat w/ additional host names: login (any provider, not limited to OIDC) is associated w/ the DNS name you navigated to - if you login to systemlink/, then go to systemlink.example.com/, you have to log in again. Maybe we should be redirecting to a 'preferred' name to avoid this.

Reply:
I added a note about OIDC and linked to the redirect URI section of that chapter. Per the other note of multiple DNS names: I am going to leave out this detail for now .